auto_tests/TCP_test.c

```diff
@@ -811,6 +815,8 @@ static void test_tcp_connection(void)
     ck_assert_msg(send_packet_tcp_connection(tc_1, 0, (const uint8_t *)"Gentoo", 6) == -1, "could send packet.");
     ck_assert_msg(kill_tcp_connection_to(tc_2, 0) == 0, "could not kill connection to\n");
 
+    netprof_kill(mem, tcp_np);
+
     kill_tcp_server(tcp_s);
     kill_tcp_connections(tc_1);
     kill_tcp_connections(tc_2);
```

Comment:
the order here is the initialization order, even the server vs connections. should this not be reversed?